syntax = "proto2";

package opentxs.proto;
option optimize_for = LITE_RUNTIME;

import public "Bailment.proto";
import public "ConnectionInfo.proto";
import public "OutBailment.proto";
import public "PeerEnums.proto";
import public "PendingBailment.proto";
import public "Signature.proto";
import public "StoreSecret.proto";

message PeerRequest {
    optional uint32 version = 1;
    optional string id = 2;
    optional string initiator = 3;
    optional string recipient = 4;
    optional PeerRequestType type = 5;
    optional string cookie = 6;
    optional Signature signature = 7;
    oneof request {
        Bailment bailment = 8;
        OutBailment outbailment = 9;
        PendingBailment pendingbailment = 10;
        ConnectionInfo connectioninfo = 11;
        StoreSecret storesecret = 12;
    }
}
